File: src/main/kotlin/lec04/Lec04Main.java
package lec04;

public class Lec04Main {
    public static void main(String[] args) {
        final JavaMoney money1 = new JavaMoney(1_000L);
        final JavaMoney money2 = new JavaMoney(2_000L);
        System.out.println(money1.plus(money2).toString());
    }
}

File: src/main/kotlin/lec04/Lec04Main.kt
package lec04

fun main() {
    val money1 = JavaMoney(1_000L)
    val money2 = money1
    val money3 = JavaMoney(1_000L)
    println(money1 === money2)
    println(money1 === money3)
    println(money2 === money3)

    println(money1 == money2)
    println(money1 == money3)
    println(money2 == money3)
}
